File: Apresentação 3/Perl/algoritmo_005.pl
print "Introduza numero do factorial a calcular:\n";
$i = <STDIN>;
$j = factorial($i);
print "Factorial de $i e: $j\n";

sub factorial
{
	my $k = shift;
	if ($k > 2)
	{
		$j = $k * factorial($k-1);
	}
	else
	{
		$j = $k;
	}
}

File: Apresentação 3/Ruby/algoritmo_005.rb
def factorial(k)
  if k > 2
    j = k * factorial(k - 1)
  else
    j = k
  end
end

puts "Introduza o número do factorial a calcular"
i = gets
print "Factorial de #{i.to_i} é: #{factorial(i.to_i).to_s}"

